vector-im/element-x-android
# Update the chat screen UI using RoomInfo.

This is specially useful for getting live values for `hasRoomCall`.
 
## Type of change


## Content

- Adds `roomInfoFlow` to `MatrixRoom` to get `MatrixRoomInfo` updates in real time.
- Use that to get if a room has an active call.
- Customize the join call button using this info.
- Also map room name and avatar url from this `RoomInfo` to get live updates.

## Motivation and context

Implements most of https://github.com/vector-im/element-x-android/issues/1158.

## Screenshots / GIFs

In the PR.

## Tests

You can take a look at the screenshots or:

- Log into 2 clients with EC integration.
- Go to the same room on both.
- Create a call in one of them, then actually join the call inside the webview.
- In the other device, the 'create call' icon should have changed to a 'Join call' button.

## Tested devices

- OS version(s): 11, 14.

## Checklist
